packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx

```diff
@@ -0,0 +1,393 @@
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 0 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> },
+          { title: 'Tab two', eventKey: 2, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  userEvent.click(screen.getByRole('tab', { name: 'More' }));
+
+  userEvent.click(document.body);
+
+  expect(screen.queryByText('Tab two')).not.toBeInTheDocument();
+});
+
+test('Calls the onTabClick callback provided via context when a tab is clicked', () => {
+  const mockHandleTabClick = jest.fn();
+
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, handleTabClick: mockHandleTabClick }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  userEvent.click(screen.getByRole('tab', { name: 'More' }));
+
+  // rather than actually clicking a menu item here we click the Select button provided via our Menu mock to keep the mocks simpler
+  userEvent.click(screen.getByRole('button', { name: 'Select' }));
+
+  expect(mockHandleTabClick).toHaveBeenCalledTimes(1);
+  expect(mockHandleTabClick).toHaveBeenCalledWith('mouseEvent', 1, 'fakeRef');
+});
+
+test('Does not show the overflowing tab count by default', () => {
+  render(
+    <OverflowTab
+      overflowingTabs={[
+        { title: 'Tab one', eventKey: 1 },
+        { title: 'Tab two', eventKey: 2 }
+      ]}
+    />
+  );
+
+  expect(screen.queryByRole('tab', { name: 'More (2)' })).not.toBeInTheDocument();
+});
+
+test('Shows the overflowing tab count when showTabCount is true', () => {
+  render(
+    <OverflowTab
+      overflowingTabs={[
+        { title: 'Tab one', eventKey: 1 },
+        { title: 'Tab two', eventKey: 2 }
+      ]}
+      showTabCount
+    />
+  );
+
+  expect(screen.getByRole('tab', { name: 'More (2)' })).toBeVisible();
+});
+
+test('Uses "More" as the default overflow tab title', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveTextContent('More');
+});
+
+test('Uses the selected tab title as the overflow tab title rather than the default title when a tab is selected', () => {
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 1 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  expect(screen.getByRole('tab')).toHaveTextContent('Tab one');
+});
+
+test('Uses custom overflow tab title text provided via defaultTitleText', () => {
+  render(<OverflowTab defaultTitleText="Test" />);
```

Comment:
Should this prop actually be called `titleText` instead? Since we're passing a "custom" value to something called "default".

Reply:
Spoke about this in slack, I see how the naming is justified to avoid the possible implication that a selected overflow item's name won't take precedence over this `defaultTitleText`. We might also consider `initial` as a keyword, but I'm ok with what it is.